### ISSUE TITLE
Problems using this plugin

### ISSUE THREAD

**Issue by Andream98** [dereuromark/cakephp-ajax#43]
Hello, I've recently downloaded this plugin to have its AJAX candy in my CakePHP 4.3 app, but I cannot manage to make it work like in the CakePHP Sandbox app's examples.

Here's what I've done up until now:

1. I've installed the plugin via composer
2. I've included it in the app by loading it inside my src/Application.php::bootstrap() method
3. I've loaded the Ajax.AjaxComponent inside the controller and whitelisted a single delete method

```
...

public function initialize(): void
{
	parent::initialize();

	if (in_array($this->request->getParam('action'), ['delete'])) {
		$this->loadComponent('Ajax.Ajax');
	}
}

...
```
4. I've created a button inside a table's rows that deletes the row that it's in via a JS fetch() call to the delete method mentioned above, here is its code (I converted the example jQuery to vanilla JS):
```
document.querySelectorAll('[data-ajax-delete="true"]').forEach((deleteButton) => {
  document.querySelector(`#${deleteButton.id}`).removeAttribute("onclick");
  deleteButton.addEventListener("click", async (event) => {
        event.preventDefault();

        let confirmMessage = deleteButton.confirm;
        if (confirmMessage && !confirm(confirmMessage)) {
          return false;
        }

        let form = deleteButton.previousElementSibling,
          url = form.action + ".json",
          tr = deleteButton.closest("tr");

        const response = await fetch(url, {
          method: "DELETE",
          headers: {
            "Content-type": "application/x-www-form-urlencoded",
          },
        });

        response.json().then(data => {
          console.log(data)
        })

        return false;
      });
});
```

But the result is not what I expected, as there are a few problems that I don't know if are related to a bad implementation on my part. In fact when the button is pressed, and the delete method gets called, the row gets effectively deleted but there are a few problems that I'm facing:

1. The FormProtectionComponent errors out ('_Token' was not found in request data) **only if** the delete method contains a redirect, even when I've excluded the delete method from being checked by FormProtection
```
public function beforeFilter(EventInterface $event)
{
	parent::beforeFilter($event);

	$this->FormProtection->setConfig('unlockedActions', ['delete']);
}
```
2. When I remove the redirect the error is gone but there isn't any data in the response object, this is what my delete method looks like:
```
public function delete(?string $id)
{
	$this->request->allowMethod(['delete']);
	$thingToDelete = $this->Things->get($id);

	if ($this->Things->delete($thingToDelete)) {
		$result = true;
		$this->Flash->success(__d('Crm', 'The custom variable has been deleted.'));
	} else {
		$result = false;
		$this->Flash->error(__d('Crm', 'The custom variable could not be deleted. Please, try again.'));
	}

	$this->set(compact('result'));
	$this->viewBuilder()->setOption('serialize', ['result']);
}
```

and this is the response object that I get after I press the button:

<img width="87" alt="image" src="https://user-images.githubusercontent.com/52160332/157870803-d3c14769-469a-4f44-ae8b-fc0f74c31e51.png">

as you can see, only variables that I manually set inside the delete method get set in the response, in contrary to what I've seen inside the examples where everything gets passed down (Flash, redirect etc.)

Did I mess something up?

**Comment by dereuromark** [comment]
You might need Flash plugin or alike as well if you want to also streamline such additional things.
Maybe we can clarify docs once you find out how to reproduce the sandbox examples.

**Comment by Andream98** [comment]
By Flash plugin do you mean this `Configure::write('Ajax.flashKey', 'FlashMessage');` referenced [here](https://github.com/dereuromark/cakephp-ajax/blob/master/docs/Component/Ajax.md#usage)?
Otherwise I don't know which Flash plugin I should be including

**Comment by dereuromark** [comment]
No, the sandbox is also using https://github.com/dereuromark/cakephp-flash
There could be some side effects
But what you describe should still be possible to work with only this plugin alone.

You might have to debug it a bit and maybe find a way to get it working.

**Comment by Andream98** [comment]
Alright, I'll try to work it out!
What about the FormProtectionComponent error? Did that ever occur to you?

**Comment by dereuromark** [comment]
I never used it. But since you unlocked it it should be fine
You can also test it by completely removing it for now and only adding it after you made things work.
Always try to keep the side effects minimal for the beginning and then add them one by one while checking if things are still workig.
Best to also add a controller integration test then on top for example.
Helps to prevent regressions going unnoticed.

**Comment by Andream98** [comment]
Ok, thanks a lot! I will post an update once I manage to set up everything up

**Comment by Andream98** [comment]
Turns out the problem was in how I was loading the `AjaxComponent` inside my Controller

I've switched from the docs method to this:

```
public function initialize(): void
{
  parent::initialize();

  $this->loadComponent(
    'Ajax.Ajax',
    [
      'viewClass' => 'Ajax.Ajax',
      'autoDetect' => true,
      'resolveRedirect' => true,
      'flashKey' => 'Flash.flash',
      'actions' => [
        'delete'
      ],
    ]
  );
}
```

and now it works as expected!

Unfortunately I don't have much time to dig deeper in finding the reason why calling loadComponent without the config didn't work for me, but I hope this will be helpful to somebody else!

**Comment by dereuromark** [comment]
Maybe we can fix up the docs?

**Comment by Andream98** [comment]
Yes, if you want I can help with that

**Comment by dereuromark** [comment]
Feel free to add whatever else might be useful to add
